scripts/analyze.sh

```diff
@@ -53,15 +61,31 @@ function check_for_require()
     return $?
 }
 
+function check_config_file()
+{
+    # effects: checks $1 for uncommented module to inclue from #CONFIG
+    #          If found, it will be put in zjs.conf.tmp and appended to src/Makefile
+
+    if [ -n "$CONFIG" ]; then
+        rval=$(grep -v '^#' < $CONFIG | grep $1='y\|m' | xargs)
+        if [ "$rval" ]; then
+            return 0
+        else
+            return 1
+        fi
```

Comment:
A few tricks to know here:
* grep like most utilties will return 0 when it's happy and >= 1 if it's not.
* $? gives you the return value of the last command run
* So you could skip capturing rval and just return $?
* Also, the < is unnecessary I think.

For posterity, Brian found that piping to xargs like this strips newlines. I suggest `tr -d '\n'` is a more decipherable way to do that, though. But anyway I don't think it's needed here. If you just run the grep | grep command then return $? I believe you get what you want.